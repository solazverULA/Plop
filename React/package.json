{
  "name": "nud-react",
  "version": "1.0.0",
  "private": true,
  "homepage": "https://timcreative.github.io/now-ui-dashboard-react",
  "dependencies": {
    "animate.css": "^3.6.1",
    "babel-preset-env": "^1.7.0",
    "babel-preset-react": "^6.24.1",
    "bootstrap": "4.0.0-beta",
    "chart.js": "^2.7.2",
    "history": "4.7.2",
    "jquery": "3.3.1",
    "node-sass": "^4.9.0",
    "node-sass-chokidar": "^1.3.0",
    "npm-run-all": "4.1.1",
    "perfect-scrollbar": "1.3.0",
    "react": "16.1.0",
    "react-animated-css": "^1.0.4",
    "react-chartjs-2": "2.7.0",
    "react-dom": "16.1.0",
    "react-google-maps": "9.2.2",
    "react-notification-alert": "0.0.7",
    "react-router-dom": "4.2.2",
    "react-scripts": "^1.1.4",
    "reactstrap": "5.0.0-alpha.4"
  },
  "devDependencies": {
    "babel-jest": "^22.0.4",
    "jest": "^22.0.4",
    "react-test-renderer": "^16.0.0"
  },
  "scripts": {
    "build-css": "node-sass-chokidar --include-path ./src --include-path ./node_modules src/ -o src/",
    "watch-css": "npm run build-css && node-sass-chokidar --include-path ./src --include-path ./node_modules src/ -o src/ --watch --recursive",
    "start-js": "react-scripts start",
    "start": "npm-run-all -p watch-css start-js",
    "build": "npm run build-css && react-scripts build",
    "test": "jest",
    "jest-update-snapshot": "jest --updateSnapshot",
    "eject": "react-scripts eject"
  }
}
