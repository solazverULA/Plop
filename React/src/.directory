[Dolphin]
Timestamp=2018,6,27,3,57,52
Version=3
