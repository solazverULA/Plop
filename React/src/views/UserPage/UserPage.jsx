import React from 'react';
import {
    Card, CardHeader, CardBody, Row, Col
} from 'reactstrap';

import { PanelHeader, FormInputs, CardAuthor, CardSocials } from '../../components';

import userBackground from '../../assets/img/bg5.jpg';
import userAvatar from '../../assets/img/mike.jpg';
import Cookies from 'react-cookies';
import Api from '../../api/Api/Api'
import Languaje from '../../api/translator/translator'

class User extends React.Component{

    constructor(props) {
    super(props);

    this.state = {
                  userId:Cookies.load('userId'),
                  user:{}
                  Email:"",
                  Password:"",
                  Gender:"",
                  Namecountry:"",
                  Namecities:"",
                  Zip_code:"",
                 };

                 Api._getUserForId(userId, (data)=>{
                    this.setState({user:data})
                 })
    }

    render(){
        return (
            <div>
                <PanelHeader size="sm"/>
                <div className="content">
                    <Row>
                        <Col md={8} xs={12}>
                            <Card>
                                <CardHeader>
                                    <h5 className="title">{Languaje("EditarUsuario")}</h5>
                                </CardHeader>
                                <CardBody>
                                    <form>
                                        <FormInputs
                                            ncols = {["col-md-5 pr-1" , "col-md-3 px-1" , "col-md-4 pl-1"]}
                                            proprieties = {[
                                                {
                                                    label : Languaje("CreateName"),
                                                    inputProps : {
                                                        type : "text",
                                                        disabled: false,
                                                        defaultValue: "Mi nombre"
                                                    }
                                                },
                                                {
                                                    label : "Username",
                                                    inputProps : {
                                                        type : "text",
                                                        defaultValue: "michael23"
                                                    }
                                                },
                                                {
                                                    label : "Email address",
                                                    inputProps : {
                                                        type : "email",
                                                        placeholder: "Email"
                                                    }
                                                }
                                            ]}
                                        />
                                        <FormInputs
                                            ncols = {["col-md-6 pr-1" , "col-md-6 pl-1"]}
                                            proprieties = {[
                                                {
                                                    label : "First Name",
                                                    inputProps : {
                                                        type : "text",
                                                        placeholder: "First Name",
                                                        defaultValue: "Mike"
                                                    }
                                                },
                                                {
                                                    label : "Last Name",
                                                    inputProps : {
                                                        type : "text",
                                                        placeholder: "Last Name",
                                                        defaultValue: "Andrew"
                                                    }
                                                }
                                            ]}
                                        />
                                        <FormInputs
                                            ncols = {["col-md-12"]}
                                            proprieties = {[
                                                {
                                                    label : "Address",
                                                    inputProps : {
                                                        type : "text",
                                                        placeholder: "Home Address",
                                                        defaultValue: "Bld Mihail Kogalniceanu, nr. 8 Bl 1, Sc 1, Ap 09"
                                                    }
                                                }
                                            ]}
                                        />
                                        <FormInputs
                                            ncols = {["col-md-4 pr-1" , "col-md-4 px-1" , "col-md-4 pl-1"]}
                                            proprieties = {[
                                                {
                                                    label : "City",
                                                    inputProps : {
                                                        type : "text",
                                                        defaultValue: "Bucharest",
                                                        placeholder: "City"
                                                    }
                                                },
                                                {
                                                    label : "Country",
                                                    inputProps : {
                                                        type : "text",
                                                        defaultValue: "Romania",
                                                        placeholder: "Country"
                                                    }
                                                },
                                                {
                                                    label : "Postal Code",
                                                    inputProps : {
                                                        type : "number",
                                                        placeholder: "ZIP Code"
                                                    }
                                                }
                                            ]}
                                        />
                                        <FormInputs
                                            ncols = {["col-md-12"]}
                                            proprieties = {[
                                                {
                                                    label : "About Me",
                                                    inputProps : {
                                                        type : "textarea",
                                                        rows: "4",
                                                        cols: "80",
                                                        defaultValue: "Lamborghini Mercy, Your chick she so thirsty, I'm in that two seat Lambo.",
                                                        placeholder: "Here can be your description"
                                                    }
                                                }
                                            ]}
                                        />
                                    </form>
                                </CardBody>
                            </Card>
                        </Col>
                        <Col md={4}  xs={12}>
                            <Card className="card-user">
                                <div className="image">
                                    <img src={userBackground} alt="..."/>
                                </div>
                                <CardBody>
                                    <CardAuthor
                                        avatar={userAvatar}
                                        avatarAlt="..."
                                        title="Mike Andrew"
                                        description="michael23"
                                    />
                                    <p className="description text-center">
                                        "Lamborghini Mercy <br/>
                                        Your chick she so thirsty <br/>
                                        I'm in that two seat Lambo"
                                    </p>
                                </CardBody>
                                <hr />
                                <CardSocials
                                    size="lg"
                                    socials={[
                                        {
                                            icon: "fab fa-facebook-f",
                                            href: "https://www.facebook.com/"
                                        },
                                        {
                                            icon: "fab fa-twitter",
                                            href: "https://www.facebook.com/"
                                        },
                                        {
                                            icon: "fab fa-google-plus-g",
                                            href: "https://plus.google.com/discover"
                                        },
                                    ]}
                                />
                            </Card>
                        </Col>
                    </Row>
                </div>
            </div>
        );
    }
}

export default User;
