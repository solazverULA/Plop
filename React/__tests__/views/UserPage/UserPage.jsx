import React from 'react';
import UserPage from '../../../src/views/UserPage/UserPage';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const loading = renderer.create(
    <UserPage/>
  ).toJSON();
  
  expect(loading).toMatchSnapshot();
});
