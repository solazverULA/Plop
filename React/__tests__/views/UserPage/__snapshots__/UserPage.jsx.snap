// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="panel-header panel-header-sm"
  />
  <div
    className="content"
  >
    <div
      className="row"
    >
      <div
        className="col-12 col-md-8"
      >
        <div
          className="card"
        >
          <div
            className="card-header"
          >
            <h5
              className="title"
            >
              Edit Profile
            </h5>
          </div>
          <div
            className="card-body"
          >
            <form>
              <div
                className="row"
              >
                <div
                  className="col-md-5 pr-1"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className=" form-control-label"
                      htmlFor={undefined}
                    >
                      Company (disabled)
                    </label>
                    <input
                      className="form-control"
                      defaultValue="Creative Code Inc."
                      disabled={true}
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="col-md-3 px-1"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className=" form-control-label"
                      htmlFor={undefined}
                    >
                      Username
                    </label>
                    <input
                      className="form-control"
                      defaultValue="michael23"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="col-md-4 pl-1"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className=" form-control-label"
                      htmlFor={undefined}
                    >
                      Email address
                    </label>
                    <input
                      className="form-control"
                      placeholder="Email"
                      type="email"
                    />
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-6 pr-1"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className=" form-control-label"
                      htmlFor={undefined}
                    >
                      First Name
                    </label>
                    <input
                      className="form-control"
                      defaultValue="Mike"
                      placeholder="First Name"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="col-md-6 pl-1"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className=" form-control-label"
                      htmlFor={undefined}
                    >
                      Last Name
                    </label>
                    <input
                      className="form-control"
                      defaultValue="Andrew"
                      placeholder="Last Name"
                      type="text"
                    />
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-12"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className=" form-control-label"
                      htmlFor={undefined}
                    >
                      Address
                    </label>
                    <input
                      className="form-control"
                      defaultValue="Bld Mihail Kogalniceanu, nr. 8 Bl 1, Sc 1, Ap 09"
                      placeholder="Home Address"
                      type="text"
                    />
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-4 pr-1"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className=" form-control-label"
                      htmlFor={undefined}
                    >
                      City
                    </label>
                    <input
                      className="form-control"
                      defaultValue="Bucharest"
                      placeholder="City"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="col-md-4 px-1"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className=" form-control-label"
                      htmlFor={undefined}
                    >
                      Country
                    </label>
                    <input
                      className="form-control"
                      defaultValue="Romania"
                      placeholder="Country"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="col-md-4 pl-1"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className=" form-control-label"
                      htmlFor={undefined}
                    >
                      Postal Code
                    </label>
                    <input
                      className="form-control"
                      placeholder="ZIP Code"
                      type="number"
                    />
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-12"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className=" form-control-label"
                      htmlFor={undefined}
                    >
                      About Me
                    </label>
                    <textarea
                      className="form-control"
                      cols="80"
                      defaultValue="Lamborghini Mercy, Your chick she so thirsty, I'm in that two seat Lambo."
                      placeholder="Here can be your description"
                      rows="4"
                    />
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        className="col-12 col-md-4"
      >
        <div
          className="card-user card"
        >
          <div
            className="image"
          >
            <img
              alt="..."
              src=""
            />
          </div>
          <div
            className="card-body"
          >
            <div
              className="author"
            >
              <a
                href="#"
              >
                <img
                  alt="..."
                  className="avatar border-gray"
                  src=""
                />
                <h5
                  className="title"
                >
                  Mike Andrew
                </h5>
              </a>
              <p
                className="description"
              >
                michael23
              </p>
            </div>
            <p
              className="description text-center"
            >
              "Lamborghini Mercy 
              <br />
              Your chick she so thirsty 
              <br />
              I'm in that two seat Lambo"
            </p>
          </div>
          <hr />
          <div
            className="button-container"
          >
            <button
              className="btn-round btn-icon btn btn-neutral btn-lg"
              href={undefined}
              onClick={[Function]}
              type={undefined}
            >
              <i
                className="fab fa-facebook-f"
              />
            </button>
            <button
              className="btn-round btn-icon btn btn-neutral btn-lg"
              href={undefined}
              onClick={[Function]}
              type={undefined}
            >
              <i
                className="fab fa-twitter"
              />
            </button>
            <button
              className="btn-round btn-icon btn btn-neutral btn-lg"
              href={undefined}
              onClick={[Function]}
              type={undefined}
            >
              <i
                className="fab fa-google-plus-g"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
